autorest/authorization.go

```diff
@@ -258,3 +261,25 @@ func (egta EventGridKeyAuthorizer) WithAuthorization() PrepareDecorator {
 	}
 	return NewAPIKeyAuthorizerWithHeaders(headers).WithAuthorization()
 }
+
+// BasicAuthorizer implements authorization for Basic authentication.
+type BasicAuthorizer struct {
+	token string
+}
+
+// NewBasicAuthorizer is
+func NewBasicAuthorizer(token string) *BasicAuthorizer {
+	return &BasicAuthorizer{token: token}
+}
+
+// WithAuthorization is
+func (ba *BasicAuthorizer) WithAuthorization() PrepareDecorator {
+	headers := make(map[string]interface{})
+	headers[authorization] = basic + " " + basicAuth(":"+ba.token)
+
+	return NewAPIKeyAuthorizerWithHeaders(headers).WithAuthorization()
+}
+
+func basicAuth(token string) string {
```

Comment:
Since this is only called in one place I'd prefer it to be folded into `WithAuthorization()`.